presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergSmoke.java

```diff
@@ -593,6 +593,210 @@ private void testPredicating(Session session, FileFormat fileFormat)
         dropTable(session, "test_predicating_on_real");
     }
 
+    @Test
+    public void testDateTransforms()
+    {
+        testWithAllFileFormats(this::testDateTransformsForFormat);
+    }
+
+    private void testDateTransformsForFormat(Session session, FileFormat format)
+    {
+        testHourTransformForFormat(session, format);
+        testDayTransformForFormat(session, format);
+        testMonthTransformForFormat(session, format);
+        testYearTransformForFormat(session, format);
+    }
+
+    private void testHourTransformForFormat(Session session, FileFormat format)
+    {
+        String select = "SELECT d_hour, row_count, d.min AS d_min, d.max AS d_max, b.min AS b_min, b.max AS b_max FROM \"test_hour_transform$partitions\"";
+
+        assertUpdate(session, format("CREATE TABLE test_hour_transform (d TIMESTAMP, b BIGINT)" +
+                " WITH (format = '%s', partitioning = ARRAY['hour(d)'])", format.name()));
+        assertUpdate(session, "INSERT INTO test_hour_transform VALUES" +
+                        "(TIMESTAMP '2015-01-01 10:01:23', 1)," +
+                        "(TIMESTAMP '2015-01-01 10:10:02', 2)," +
+                        "(TIMESTAMP '2015-01-01 10:55:00', 3)," +
+                        "(TIMESTAMP '2015-05-15 12:05:01', 4)," +
+                        "(TIMESTAMP '2015-05-15 12:21:02', 5)," +
+                        "(TIMESTAMP '2015-02-21 13:11:11', 6)," +
+                        "(TIMESTAMP '2015-02-21 13:12:12', 7)",
+                7);
+
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_hour_transform$partitions\"", "SELECT 3");
+
+        assertQuery(session, "SELECT COUNT(*) FROM test_hour_transform WHERE hour(d) = 10", "SELECT 3");
```

Comment:
What time zone are the tests using? Can you check if they're using legacy timestamps? If it's not UTC time zone and legacy timestamps, I'm surprised this works.

Reply:
Can you change all of these to use `SELECT d` so that we confirm the exact timestamps we expect are returned? Same goes for all of the tests that use `SELECT count(*)`

Reply:
The Session.timeZoneKey = "Pacific/Apia".  The session's system properties map is empty, i.e., no legacy_timestamp property.

I changed all the SELECT count(*) uses to explicitly test which rows were returned.